### ISSUE TITLE
2021 week 27: rebase off upstream

### ISSUE THREAD

**Issue by adpaco-aws** [model-checking/rmc#290]

**Comment by adpaco-aws** [comment]
No conflicts during the rebase.

**Comment by adpaco-aws** [comment]
But we get 3 errors when compiling:
```
error[E0599]: no function or associated item named `bind` found for struct `Binder` in the current scope
  --> compiler/rustc_codegen_llvm/src/gotoc/monomorphize/collector.rs:38:33
   |
38 |     let trait_ref = ty::Binder::bind(
   |                                 ^^^^ function or associated item not found in `Binder<'_, _>`

error[E0599]: no method named `recursion_limit` found for reference `&Session` in the current scope
   --> compiler/rustc_codegen_llvm/src/gotoc/monomorphize/collector.rs:304:18
    |
304 |     if !tcx.sess.recursion_limit().value_within_limit(adjusted_recursion_depth) {
    |                  ^^^^^^^^^^^^^^^ method not found in `&Session`

error[E0599]: no method named `type_length_limit` found for reference `&Session` in the current scope
   --> compiler/rustc_codegen_llvm/src/gotoc/monomorphize/collector.rs:338:18
    |
338 |     if !tcx.sess.type_length_limit().value_within_limit(type_length) {
    |                  ^^^^^^^^^^^^^^^^^ method not found in `&Session`

error: aborting due to 3 previous errors
```
Investigating now.

**Comment by adpaco-aws** [comment]
The last ones are due to a rework on the limits, which are not an attribute of the session but of the Rust typing context directly. [This commit](https://github.com/rust-lang/rust/commit/ff15b5e2c76bb5c0fdd64e49ee76fbd2023415bd) and [this commit](https://github.com/rust-lang/rust/commit/7e5a88a56cf14bf691c25cad04837e82c4c601f1) showed what changes were needed.

**Comment by adpaco-aws** [comment]
The first one is due to [this commit](https://github.com/rust-lang/rust/commit/884053a4b422e63a78eddd7b49793c32020d321b) that removes `bind` from `Binder`.

**Comment by adpaco-aws** [comment]
Candidate branch is `main-153-2021-07-09` which can be found [here](https://github.com/model-checking/rmc/tree/main-153-2021-07-09). CI fails since the fixes have been submitted for review in #318

**Comment by adpaco-aws** [comment]
#318 was merged and `main-153-2021-07-09` is now the default branch.
